Crash reports when adding PWAs

#### General Information

* **OpenLauncher Version:** Current nightly (0f7fc2272e1e939dfe263dd8eb99ef3007bac9d6)
* **Android Version:** 7.1.2
* **Custom ROM:** Lineage
* **Install Source:** APK copied out of build artifact

#### Description

Adding a PWA launcher from Chromium sometimes produces a crash screen:

#### Log
A crash report was shown:

> Build version: 0.7.4 
> Build date: 1980-01-01 00:00:00 
> Current date: 2022-09-01 08:33:01 
> Device: Sony E5823 
>  
> Stack trace:  
> java.lang.RuntimeException: Unable to start receiver com.benny.openlauncher.receivers.ShortcutReceiver: java.lang.RuntimeException: Setup has not been initialised!
> 	at android.app.ActivityThread.handleReceiver(ActivityThread.java:3072)
> 	at android.app.ActivityThread.-wrap18(ActivityThread.java)
> 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1580)
> 	at android.os.Handler.dispatchMessage(Handler.java:102)
> 	at android.os.Looper.loop(Looper.java:154)
> 	at android.app.ActivityThread.main(ActivityThread.java:6186)
> 	at java.lang.reflect.Method.invoke(Native Method)
> 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889)
> 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)
> Caused by: java.lang.RuntimeException: Setup has not been initialised!
> 	at com.benny.openlauncher.manager.Setup.get(Setup.java:25)
> 	at com.benny.openlauncher.manager.Setup.appLoader(Setup.java:47)
> 	at com.benny.openlauncher.receivers.ShortcutReceiver.onReceive(ShortcutReceiver.java:54)
> 	at android.app.ActivityThread.handleReceiver(ActivityThread.java:3065)
> 	... 8 more

### Further references

PWA shortcuts were introduced in #637 by @Apollo3zehn -- if you read this and it does ring a bell, hints would be appreciated.

### Next steps

I'll yet have to see whether I can make anything of this or fix it myself.

--- a/app/src/main/java/com/benny/openlauncher/activity/homeparts/HpInitSetup.java
+++ b/app/src/main/java/com/benny/openlauncher/activity/homeparts/HpInitSetup.java
@@ -2,10 +2,8 @@ package com.benny.openlauncher.activity.homeparts;
 
 import android.content.Context;
 import android.support.annotation.NonNull;
-import android.util.Log;
 
 import com.benny.openlauncher.AppObject;
-import com.benny.openlauncher.activity.HomeActivity;
 import com.benny.openlauncher.manager.Setup;
 import com.benny.openlauncher.util.AppManager;
 import com.benny.openlauncher.util.AppSettings;
@@ -19,11 +17,11 @@ public final class HpInitSetup extends Setup {
     private final HpEventHandler _eventHandler;
     private final AppSettings _appSettings;
 
-    public HpInitSetup(HomeActivity homeActivity) {
+    public HpInitSetup(Context context) {
         _appSettings = AppSettings.get();
         _desktopGestureCallback = new HpGestureCallback(_appSettings);
-        _dataManager = new DatabaseHelper(homeActivity);
-        _appLoader = AppManager.getInstance(homeActivity);
+        _dataManager = new DatabaseHelper(context.getApplicationContext());
+        _appLoader = AppManager.getInstance(context.getApplicationContext());
         _eventHandler = new HpEventHandler();
     }
 

--- a/app/src/main/java/com/benny/openlauncher/receivers/ShortcutReceiver.java
+++ b/app/src/main/java/com/benny/openlauncher/receivers/ShortcutReceiver.java
@@ -13,6 +13,7 @@ import android.util.Log;
 
 import com.benny.openlauncher.R;
 import com.benny.openlauncher.activity.HomeActivity;
+import com.benny.openlauncher.activity.homeparts.HpInitSetup;
 import com.benny.openlauncher.manager.Setup;
 import com.benny.openlauncher.model.App;
 import com.benny.openlauncher.model.Item;
@@ -49,6 +50,10 @@ public class ShortcutReceiver extends BroadcastReceiver {
         } finally {
             if (shortcutIcon == null)
                 shortcutIcon = new BitmapDrawable(context.getResources(), (Bitmap) intent.getExtras().getParcelable(Intent.EXTRA_SHORTCUT_ICON));
+        }
+
+        if (!Setup.wasInitialised()) {
+            Setup.init(new HpInitSetup(context));
         }
 
         App app = Setup.appLoader().createApp(shortcutIntent);

